Added Nod Covert Operations mission: scb30ea - Eviction Notice

YAmismo opened the pull request:
Hi again, i've been porting this Nod map while learning how to do it. I hope its ready and good enough, at least i had fun doing it.

abcdefg30 commented:
Hi. I haven't looked at the changes yet, but there are some lint errors reported by the utility:
```
OpenRA.Utility(1,1): Warning: The player Neutral owning the world should not have any allies or enemies.
OpenRA.Utility(1,1): Error: rules.yaml:108 refers to a trait field `Prerequisites` that does not exist on `ResourceValueMultiplier`.
OpenRA.Utility(1,1): Error: rules.yaml:112 refers to a trait field `Prerequisites` that does not exist on `ResourceValueMultiplier`.
```

YAmismo commented:
Can i run this lint test myself on my pc? Just to check before trying to push on every change.

PunkPun commented:
Yes, you can run `make test` in the project directory

YAmismo commented:
Well, test doesn't show any error for me now.

abcdefg30 commented on mods/cnc/maps/scb30ea/rules.yaml:
```diff
@@ -0,0 +1,139 @@
+
+CAMERA:
+	RevealsShroud:
+		Range: 5c0
```
This and the other files will want a new line at the end.

abcdefg30 commented on mods/cnc/maps/scb30ea/rules.yaml:
```diff
@@ -0,0 +1,139 @@
+		Sequence: dollar
+		UseCashTick: true
+
+# Initial flame tanks cant move with their own husks
```
```suggestion
# Initial flame tanks can't move due to their own husks
```
Or remove that comment.

abcdefg30 commented on mods/cnc/maps/scb30ea/scb30ea-AI.lua:
```diff
@@ -0,0 +1,288 @@
+
```
Missing header.

YAmismo replied:
Sorry, not sure of this... Is the copyright comment what you refer to?

abcdefg30 replied:
Yes. The same thing all the other lua files have.

abcdefg30 commented on mods/cnc/maps/scb30ea/scb30ea-AI.lua:
```diff
@@ -0,0 +1,288 @@
+
+
+GDInuke1 = {type = "nuk2", pos = CPos.New(6,3), blocked = false}
```
Our code style is to have a space after and before braces.
```suggestion
GDInuke1 = { type = "nuk2", pos = CPos.New(6,3), blocked = false }
```

abcdefg30 commented on mods/cnc/maps/scb30ea/scb30ea-AI.lua:
```diff
@@ -0,0 +1,288 @@
+	GDInuke4, GDIdef5, GDIdef6, GDInuke5, GDIsilo1, GDIsilo2, GDIdef7, GDIdef8, GDIsilo3, GDIsilo4, GDIdef9}
+BSizes = { nuk2 = CVec.New(2,3), proc = CVec.New(3,4), pyle = CVec.New(2,3), weap = CVec.New(3,3), atwr = CVec.New(1,1), gtwr = CVec.New(1,1), silo = CVec.New(2,1), hpad = CVec.New(2,2)}
+
+ProductionBuildings = {i = GDIPyle ,v = GDIWeap, a = GDICYard}
```
Space should be after the comma
```suggestion
ProductionBuildings = {i = GDIPyle, v = GDIWeap, a = GDICYard}
```

abcdefg30 commented on mods/cnc/maps/scb30ea/scb30ea-AI.lua:
```diff
@@ -0,0 +1,288 @@
+
+	RepairNamedActors(GDI, 0.75)
+
+	if Difficulty == "hard" then PBDelay = 115 PUDelay = 30 ICDelay = 800 ProdCDSecs = 30 GDI.Resources = 10000 minHarvs = 3 ProcUpgrade = "AIHProcUpgrade" end
```
Please move these onto own lines
```
if Difficulty == "hard" then
	[...]
end
```

abcdefg30 commented on mods/cnc/maps/scb30ea/scb30ea-AI.lua:
```diff
@@ -0,0 +1,288 @@
+		Trigger.AfterDelay(DateTime.Seconds(ICDelay), function()
+			FireIonCannon(300)
+		end)
+		Media.DisplayMessage("The GDI are preparing their ion cannon. Don't let them get used to it.")
```
Also, this needs to account for #20265 now.

YAmismo replied:
Can't test this, but comparing to the other maps, i think its correct. (Do i need to compile full project? I guess the last playtest doesn't support it right?)

abcdefg30 replied:
It does look correct on a quick glance, I'll test it later. Yeah the latest playtest doesn't have all the features; generally it is better to compile and use the latest dev/bleed version.

abcdefg30 commented on mods/cnc/maps/scb30ea/scb30ea-AI.lua:
```diff
@@ -0,0 +1,321 @@
+		Nod.Resources = 3000
+	end
+
+	if ProcUpgrade then ProcUpg = Actor.Create(ProcUpgrade, true, { Owner = GDI } ) end
```
Ah this is my bad, sorry. The line I was commenting on was an assignment and there we have a space before and after the braces, but there is no space between them and parenthesis:
```suggestion
	if ProcUpgrade then
		ProcUpg = Actor.Create(ProcUpgrade, true, { Owner = GDI })
	end
```
The other usages seem fine, so I'll quickly mark the places I spotted so you don't have to go over them again.

abcdefg30 commented on mods/cnc/maps/scb30ea/scb30ea-AI.lua:
```diff
@@ -0,0 +1,321 @@
+			end
+		end
+	end
+	if not CheckProgrammed then CheckProgrammed = true Trigger.AfterDelay(250, function() CheckProgrammed = false CheckBase() end) end
```
```suggestion
	if not CheckProgrammed then
		CheckProgrammed = true
		Trigger.AfterDelay(250, function()
			CheckProgrammed = false
			CheckBase()
		end)
	end
```

abcdefg30 commented on mods/cnc/maps/scb30ea/scb30ea-AI.lua:
```diff
@@ -0,0 +1,321 @@
+			return
+		end
+
+		local newbuilding = Actor.Create(building.type, true, { Owner = GDI, Location = building.pos } )
```
```suggestion
		local newbuilding = Actor.Create(building.type, true, { Owner = GDI, Location = building.pos })
```

abcdefg30 commented on mods/cnc/maps/scb30ea/scb30ea-AI.lua:
```diff
@@ -0,0 +1,321 @@
+		ProductionBuildings["v"].Build( { "harv" }, function()
+			CheckProduction()
+			Trigger.AfterDelay(DateTime.Seconds(5), function()
+				if not ProductionBuildings["v"].IsDead then ProductionBuildings["v"].Build( { "harv" } ) end
```
```suggestion
				if not ProductionBuildings["v"].IsDead then
					ProductionBuildings["v"].Build( { "harv" } )
				end
```

abcdefg30 commented on mods/cnc/maps/scb30ea/scb30ea-AI.lua:
```diff
@@ -0,0 +1,321 @@
+--[[
+   Copyright 2007-2022 The OpenRA Developers (see AUTHORS)
```
This line changed compared to the last released version too:
```suggestion
   Copyright (c) The OpenRA Developers and Contributors
```

abcdefg30 commented on mods/cnc/maps/scb30ea/scb30ea.lua:
```diff
@@ -0,0 +1,329 @@
+--[[
+   Copyright 2007-2022 The OpenRA Developers (see AUTHORS)
```
```suggestion
   Copyright (c) The OpenRA Developers and Contributors
```

abcdefg30 commented on mods/cnc/maps/scb30ea/scb30ea.lua:
```diff
@@ -0,0 +1,329 @@
+			end
+		end)
+		waahTriggered = true
+		cam = Actor.Create("camera", true, { Owner = Nod, Location = CPos.New(53,44) } )
```
```suggestion
		cam = Actor.Create("camera", true, { Owner = Nod, Location = CPos.New(53,44) })
```

abcdefg30 commented on mods/cnc/maps/scb30ea/scb30ea.lua:
```diff
@@ -0,0 +1,329 @@
+end
+
+StartGuard = function()
+	Trigger.OnAllKilled(deft1, function() table.insert(TQueue, { t = DefenseTeams, j = "defend" } ) end)
```
```suggestion
	Trigger.OnAllKilled(deft1, function() table.insert(TQueue, { t = DefenseTeams, j = "defend" }) end)
```

abcdefg30 commented on mods/cnc/maps/scb30ea/scb30ea.lua:
```diff
@@ -0,0 +1,329 @@
+end
+
+StartPatrol = function()
+	Trigger.OnAllKilled(patrol1, function() table.insert(TQueue, { t = PatrolTeams, j = "patrol" } ) end)
```
```suggestion
	Trigger.OnAllKilled(patrol1, function() table.insert(TQueue, { t = PatrolTeams, j = "patrol" }) end)
```

abcdefg30 commented on mods/cnc/maps/scb30ea/scb30ea.lua:
```diff
@@ -0,0 +1,329 @@
+					else
+						i = i + 1
+					end
+					Trigger.AfterDelay(20, function() MoveAsGroup( team, path, i, loop) end)
```
```suggestion
					Trigger.AfterDelay(20, function()
						MoveAsGroup(team, path, i, loop)
					end)
```

abcdefg30 commented on mods/cnc/maps/scb30ea/scb30ea.lua:
```diff
@@ -0,0 +1,329 @@
+	Nod = Player.GetPlayer("Nod")
+	GDI = Player.GetPlayer("GDI")
+	Camera.Position = PlayerStart.CenterPosition
+	Flare = Actor.Create("flare", true, { Owner = Nod, Location = DefaultFlareLocation.Location } )
```
```suggestion
	Flare = Actor.Create("flare", true, { Owner = Nod, Location = DefaultFlareLocation.Location })
```

abcdefg30 commented on mods/cnc/maps/scb30ea/scb30ea.lua:
```diff
@@ -0,0 +1,329 @@
+	InitObjectives(Nod)
+	EliminateAllGDI = AddPrimaryObjective(Nod, "eliminate-gdi-forces")
+	FindAllCivMoney = AddSecondaryObjective(Nod, "take-civilians-money-crates")
+-- DestroyIonCannon objective in ai section
```
I think you can remove these comments, we don't usually leave direct references to what is equivalent to what in the ini file(s).

abcdefg30 commented on mods/cnc/maps/scb30ea/scb30ea.lua:
```diff
@@ -0,0 +1,329 @@
+
+--Argh trigger, reinforce with team grnf2 engineers
+	Trigger.OnAllKilled( Civsargh, function()
+		local cargo = Reinforcements.ReinforceWithTransport(GDI, "tran", grnf2, { CPos.New(0,32), waypoint10.Location }, { CPos.New(0,32) } )[2]
```
```suggestion
		local cargo = Reinforcements.ReinforceWithTransport(GDI, "tran", grnf2, { CPos.New(0,32), waypoint10.Location }, { CPos.New(0,32) })[2]
```

abcdefg30 commented on mods/cnc/maps/scb30ea/weapons.yaml:
```diff
@@ -0,0 +1,3 @@
+IonCannon:
+	Warhead@1Dam_impact: SpreadDamage
+		Falloff: 1000, 60, 40, 20
```
Please add a new line at the end of this file as well.

abcdefg30 reviewed (commented):
First observations:
- There is a single civilian in the village which isn't hostile (no attack cursor) and has the "Unit Lost" notification playing when dying. Is that intentional?
- The objective to collect all money didn't complete for me
- Directly sending two mammoth tanks into the base on normal difficulty seems a bit harsh to me. (I assume that is `grnf3`? It would be nice if those were a bit more descriptive names. I can help with that later though.)

YAmismo commented:
> There is a single civilian in the village which isn't hostile (no attack cursor) and has the "Unit Lost" notification playing when dying. Is that intentional?

No, he was there in the original map, simply i didn't change that. Maybe he's there to throw the flare? Like a spy possibly.

> The objective to collect all money didn't complete for me

There are 5 crates, did you found it? I managed to build and run the dev_version and it works for me.

> Directly sending two mammoth tanks into the base on normal difficulty seems a bit harsh to me. (I assume that is grnf3? It would be nice if those were a bit more descriptive names. I can help with that later though.)

It's grnf1 (i'll change few names like that for more descriptive ones).
The team are triggered by destroying certain civilians buildings, (only 5 of them), so you are right its too random when the player doesn't know why this happens. Maybe taking more buildings to be triggered... or giving a warning while doing so.
Making it trigger by destoyed buildings count (different number on difficulty), instead of specific ones... could work. I'll try this later.

PunkPun commented:
Hello, could you squash the commits?

YAmismo commented:
Sorry, can't understand what you mean by squash, my english is far from perfect. Can you explain?
It's ok, just discovered what is it.

YAmismo commented:
Sorry again, i cant figure out how to do it. I'll try again later.

YAmismo committed:
Added Nod Covert Operations - Eviction Notice - scb30ea

abcdefg30 commented:
Sorry for the long delay in answering.

> No, he was there in the original map, simply i didn't change that. Maybe he's there to throw the flare? Like a spy possibly.

I like the idea. We should definitely either do something with the civ or change its owner. As it is now it just feels weird Imo, and gives you a "Unit Lost" notification.

> There are 5 crates, did you found it? I managed to build and run the dev_version and it works for me.

My mistake, I didn't notice the crate spawned by the single house in the north.

> It's grnf1 (i'll change few names like that for more descriptive ones).
The team are triggered by destroying certain civilians buildings, (only 5 of them), so you are right its too random when the player doesn't know why this happens. Maybe taking more buildings to be triggered... or giving a warning while doing so.
Making it trigger by destoyed buildings count (different number on difficulty), instead of specific ones... could work. I'll try this later.

That works quite good. I still think the mammoths are a bit strong, but now it's not random any more and should be predictable by the player.

YAmismo commented:
> Sorry for the long delay in answering.

No problem at all.

Maybe change the civilian for a minigunner? For the flare story. Or better remove it?
What you think?

YAmismo commented:
Replaced the civilian with a minigunner, it's more clear now that a nod unit was there, waiting reinforcements.

Mailaender commented on mods/cnc/maps/scb30ea/scb30ea.lua:
```diff
@@ -0,0 +1,376 @@
+CivsMoneyBuildings = { Actor37, Actor41, Actor42, Actor44, Actor58 }
+
+ReiforcementsMammoths = { "htnk", "htnk" }
+ReiforcementsEngineers = { "e6", "e6", "e6", "e6", "e6" }
```
```suggestion
ReinforcementsMammoths = { "htnk", "htnk" }
ReinforcementsEngineers = { "e6", "e6", "e6", "e6", "e6" }
```

Mailaender commented on mods/cnc/maps/scb30ea/scb30ea-AI.lua:
```diff
@@ -0,0 +1,421 @@
+
+
+SendAttackGroup = function(team)
+	sendteam = team
```
```suggestion
	SendTeam = team
```

Uppercase globals

Mailaender commented on mods/cnc/maps/scb30ea/scb30ea-AI.lua:
```diff
@@ -0,0 +1,421 @@
+
+RestartUnitProduction = function()
+	if not restarting then
+		restarting = true
```
```suggestion
		Restarting = true
```

Uppercase globals.

Mailaender commented on mods/cnc/maps/scb30ea/scb30ea.lua:
```diff
@@ -0,0 +1,376 @@
+	end)
+
+
+	Trigger.OnAllKilled( Civilians, function()
```
```suggestion
	Trigger.OnAllKilled(Civilians, function()
```

Mailaender commented on mods/cnc/maps/scb30ea/scb30ea.lua:
```diff
@@ -0,0 +1,376 @@
+	end)
+
+
+	Utils.Do( RunForHelpCivs, function(actor)
+		Trigger.OnDiscovered( actor, function()
```
```suggestion
	Utils.Do(RunForHelpCivs, function(actor)
		Trigger.OnDiscovered(actor, function()
```

and the plenks in between.

Mailaender commented on mods/cnc/maps/scb30ea/scb30ea.lua:
```diff
@@ -0,0 +1,376 @@
+			Trigger.OnIdle(a, function()
+				Trigger.Clear(a, "OnIdle")
+				a.Stance = "Defend"
+				local ii = 0
```
```suggestion
				local teamNumber = 0
```
?

Mailaender commented on mods/cnc/maps/scb30ea/scb30ea-AI.lua:
```diff
@@ -0,0 +1,421 @@
+
+ProductionBuildings = { i = GDIPyle, v = GDIWeap, a = GDICYard }
+
+ProdQueue = { i = { }, v = { }, a = { } }
```
```suggestion
ProductionQueue = { infantry = { }, vehicle = { }, aircraft = { } }
```

maybe? I had a hard time reading this.

Mailaender commented on mods/cnc/maps/scb30ea/scb30ea.lua:
```diff
@@ -0,0 +1,376 @@
+	if CivsBuildingsToDestroy == 2 then
+		Media.DisplayMessage(UserInterface.Translate("village-destruction-warning"))
+	elseif CivsBuildingsToDestroy == 0 then
+		local team = Reinforcements.Reinforce(GDI, ReiforcementsMammoths, { CPos.New(2,9), CPos.New(3,9) }, 40, function(a)
```
```suggestion
		Reinforcements.Reinforce(GDI, ReiforcementsMammoths, { CPos.New(2,9), CPos.New(3,9) }, 40, function(a)
```
Unused local.

YAmismo commented:
Done.

penev92 commented:
@YAmismo can you confirm which mission this is? #4988 lists `scb30ea` as "N64 Special Ops - Nod", while "CovertOperations - Nod" are `scb21-28` 🤔

YAmismo commented:
I extracted (with xcc mixer) and converted scb30ea.ini/bin from sc-001.mix.

According to https://cnc.fandom.com/wiki/Eviction_Notice_(Covert_Operations) and the new c&c remastered, its scb30ea Nod Covert Operations.

Don't know where #4988 list came from, but it does not coincide with any list i can find.

penev92 commented:
OK, thanks. Then we may need to fix #4988

YAmismo commented:
>ProdQueue = { i = { }, v = { }, a = { } }
ProductionQueue = { infantry = { }, vehicle = { }, aircraft = { } }

Just fixed a bug i accidentally introduced while renaming this.

JovialFeline commented on mods/cnc/maps/scb30ea/scb30ea.lua:
```diff
@@ -0,0 +1,376 @@
+		end)
+		CivsRunning = true
+		local cam = Actor.Create("camera", true, { Owner = Nod, Location = CPos.New(53,44) })
+		Trigger.AfterDelay(125, function() cam.Destroy() end)
```
```suggestion
		Trigger.AfterDelay(125, cam.Destroy)
```

JovialFeline replied:
There are some Trigger lines that could be shortened, where a single function is called without arguments. This already seems to be done elsewhere, and more consistency can't hurt.
```suggestion
		Trigger.AfterDelay(125, cam.Destroy)
```

JovialFeline commented on mods/cnc/languages/lua/en.ftl:
```diff
@@ -140,3 +140,12 @@ destroy-tech-center = Destroy the GDI R&D center.
+
+## nod30
+take-civilians-money-crates = Find all the civilians money.
+    They won't need it anymore.
```
"The civilians money" doesn't seem quite right. "The civilians' money" or even "the civilian money" ought to work.

JovialFeline commented on mods/cnc/languages/lua/en.ftl:
```diff
@@ -140,3 +140,12 @@ destroy-tech-center = Destroy the GDI R&D center.
+take-civilians-money-crates = Find all the civilians money.
+    They won't need it anymore.
+quickly-destroy-ion-cannon = Destroy GDI Ion Cannon before
+    it fires two times.
```
This could be a personal thing, but I had a moment of confusion about this in-game. I then realized it meant destroying/capturing the communications center guiding the cannon. Perhaps "disrupt" or "disable" would fit better?

JovialFeline commented on mods/cnc/languages/lua/en.ftl:
```diff
@@ -140,3 +140,12 @@ destroy-tech-center = Destroy the GDI R&D center.
+    They won't need it anymore.
+quickly-destroy-ion-cannon = Destroy GDI Ion Cannon before
+    it fires two times.
+civilians-runs = Soldier: Hey, those civilians... where are they going?
```
This message could be tweaked for less awkwardness. `Media.DisplayMessage` can specify a different prefix or "speaker" in its second argument. [I changed the .ftl and .lua for a before/after demonstration here.](https://i.imgur.com/mOIAXR6.png)
`gdi04a` or `gdi04b` have an example of this, though `allies03a` from RA is more straightforward:
```lua
Media.DisplayMessage(UserInterface.Translate("tanya-rules-of-engagement"), UserInterface.Translate("tanya"))
```

JovialFeline commented on mods/cnc/languages/lua/en.ftl:
```diff
@@ -140,3 +140,12 @@ destroy-tech-center = Destroy the GDI R&D center.
+    it fires two times.
+civilians-runs = Soldier: Hey, those civilians... where are they going?
+destroy-ion-cannon-advise = The GDI are preparing their ion cannon. Don't let them get used to it.
+village-destruction-warning = Be careful commander, the GDI won't stand still while we burn the entire village.
```
```suggestion
village-destruction-warning = Be careful, commander. The GDI won't stand still while we burn the entire village.
```

JovialFeline commented on mods/cnc/maps/scb30ea/scb30ea.lua:
```diff
@@ -0,0 +1,376 @@
+	Flare = Actor.Create("flare", true, { Owner = Nod, Location = DefaultFlareLocation.Location })
+	Trigger.AfterDelay(DateTime.Minutes(1), function()
+		Flare.Destroy()
+	end)
```
```suggestion
	Trigger.AfterDelay(DateTime.Minutes(1), Flare.Destroy)
```

JovialFeline commented on mods/cnc/maps/scb30ea/scb30ea.lua:
```diff
@@ -0,0 +1,376 @@
+			BombTriggered = true
+			Trigger.AfterDelay(25, function() SendGDIAirstrike() end)
+			Trigger.AfterDelay(150, function() SendGDIAirstrike() end)
+			Trigger.AfterDelay(275, function() SendGDIAirstrike() end)
```
```suggestion
			Trigger.AfterDelay(25, SendGDIAirstrike)
			Trigger.AfterDelay(150, SendGDIAirstrike)
			Trigger.AfterDelay(275, SendGDIAirstrike)
```

JovialFeline commented on mods/cnc/maps/scb30ea/scb30ea.lua:
```diff
@@ -0,0 +1,376 @@
+
+
+	Utils.Do(CiviliansBuildings, function(b)
+		Trigger.OnKilled(b, function() CheckVillageDestruction() end)
```
```suggestion
		Trigger.OnKilled(b, CheckVillageDestruction)
```

JovialFeline commented on mods/cnc/maps/scb30ea/scb30ea.lua:
```diff
@@ -0,0 +1,376 @@
+	Utils.Do(OilDerricks, function(actor)
+		Trigger.OnKilledOrCaptured(actor, function()
+			OilDerricksAirstrike()
+		end)
```
```suggestion
		Trigger.OnKilledOrCaptured(actor, OilDerricksAirstrike)
```

JovialFeline commented on mods/cnc/maps/scb30ea/scb30ea.lua:
```diff
@@ -0,0 +1,376 @@
+	Utils.Do(RunForHelpCivs, function(actor)
+		Trigger.OnDiscovered(actor, function()
+			RunForHelp()
+		end)
```
```suggestion
		Trigger.OnDiscovered(actor, RunForHelp)
```

JovialFeline commented on mods/cnc/maps/scb30ea/scb30ea.lua:
```diff
@@ -0,0 +1,376 @@
+		if a.Owner == Nod then
+			Utils.Do(WelcomeTeam, function(a)
+				if not a.IsDead then
+					a.AttackMove(PlayerStart.Location) IdleHunt(a)
```
```suggestion
					a.AttackMove(PlayerStart.Location)
					IdleHunt(a)
```

JovialFeline commented on mods/cnc/maps/scb30ea/scb30ea-AI.lua:
```diff
@@ -0,0 +1,420 @@
+--Building logic
+SetAutoRebuild = function()
+	Utils.Do(GDIBase, function(b)
+		Trigger.OnKilled(b, function() CheckBase() end)
```
Just some more Trigger trimmings.
```suggestion
		Trigger.OnKilled(b, CheckBase)
```

JovialFeline commented on mods/cnc/maps/scb30ea/scb30ea-AI.lua:
```diff
@@ -0,0 +1,420 @@
+		end
+
+		local newbuilding = Actor.Create(building.type, true, { Owner = GDI, Location = building.pos })
+		Trigger.OnKilled(newbuilding, function() CheckBase() end)
```
```suggestion
		Trigger.OnKilled(newbuilding, CheckBase)
```

JovialFeline commented on mods/cnc/maps/scb30ea/scb30ea-AI.lua:
```diff
@@ -0,0 +1,420 @@
+ProdCooldown = false
+CheckProduction = function()
+	if #GDI.GetActorsByType("proc") < 1 and GDI.Resources < 6000 then
+		Trigger.AfterDelay(250, function() CheckProduction() end)
```
```suggestion
		Trigger.AfterDelay(250, CheckProduction)
```

JovialFeline commented on mods/cnc/maps/scb30ea/scb30ea-AI.lua:
```diff
@@ -0,0 +1,420 @@
+			RestartUnitProduction()
+		end
+
+		Trigger.AfterDelay(50, function() CheckBase() end)
```
```suggestion
		Trigger.AfterDelay(50, CheckBase)
```

JovialFeline commented on mods/cnc/maps/scb30ea/scb30ea.lua:
```diff
@@ -0,0 +1,376 @@
+	if i == 1 and not loop then
+		Utils.Do(team, function(u)
+			Trigger.OnDamaged(u, function()
+				Trigger.AfterDelay(2, function()
+					Utils.Do(team, function(u)
+						if not u.IsDead then
+							Trigger.Clear(u, "OnDamaged")
+							IdleHunt(u)
+						end
+					end)
+				end)
```
This delay seems harmless but also not necessary. If you were *adding* a trigger in the same tick, that could certainly be a problem. A similar thing is done further down with `OnIdle` (line 190-191), but that part has no delay. If this is a mistake on my part, please pardon me. It struck me as clutter in a function that already has a lot going on.

JovialFeline reviewed (changes requested):
Hello! This is more of a playtest review than a code review. I went through on Hard and Normal, after completing the original mission for comparison. I thought this was a pretty good translation to ORA, so kudos.

I did find it a bit rough to complete, but this was even more true of the original in my experience, and most of the failures were early game. That's fair enough since it's an expansion mission for a game that's routinely mean. I'll also share that the Ion Cannon zapped one of my turrets and the civilian building next to it. That prompted a warning about wrecking the village, which I found amusing. This is not a critique; no change is needed!

I have not had issues with how the script works and I don't intend to suggest large changes. However, I do have suggestions for easy polish if you can indulge me.

#20743 will get a look from me once I can manage the original. The early game has been a doozy, and then I learned Nod has nukes on that map. Huzzah.

YAmismo commented:
Nice review, thanks. I'll take a look on all of this.
Those translation issues... haha my english still needs a lot of polish.

YAmismo commented:
Updated with suggested changes by JovialFeline, renamed map files and capture structures function improved.

Smittytron commented on mods/cnc/maps/eviction-notice/eviction-notice.lua:
```diff
@@ -0,0 +1,374 @@
+GDIBaseEntranceCells = { CPos.New(16,5), CPos.New(16,6), CPos.New(16,7), CPos.New(16,8), CPos.New(17,4), CPos.New(17,5), CPos.New(17,6), CPos.New(17,7), CPos.New(17,8), CPos.New(18,4), CPos.New(18,5), CPos.New(19,4), CPos.New(8,15), CPos.New(9,15), CPos.New(10,15), CPos.New(11,15), CPos.New(12,15), CPos.New(13,15), CPos.New(14,15), CPos.New(9,16), CPos.New(10,16), CPos.New(11,16), CPos.New(12,16) ...
+InnerGDIBaseEntranceCells = { CPos.New(7,5), CPos.New(7,6), CPos.New(7,7), CPos.New(8,6), CPos.New(8,7), CPos.New(9,6), CPos.New(9,7) }
+
+
```
For consistency with our other missions one space between functions please.

Smittytron commented on mods/cnc/maps/eviction-notice/eviction-notice.lua:
```diff
@@ -0,0 +1,374 @@
+		Nod.MarkCompletedObjective(DestroyIonCannon)
+		PrepareOrcas()
+	end)
+
```
Remove space

Smittytron commented on mods/cnc/maps/eviction-notice/eviction-notice.lua:
```diff
@@ -0,0 +1,374 @@
+	FindAllCivMoney = AddSecondaryObjective(Nod, "take-civilians-money-crates")
+	CheckObjectives()
+	InitAi()
+
```
One space between triggers as well.

Smittytron reviewed (commented):
Code LGTM sans a few spacing nits.

YAmismo commented:
Spaces removed :wink:

Smittytron commented:
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/49d499506675ca364943ab31207d58864c034a98)
